### PR TITLE
Fixing bug in auto enrolment for subject knowledge

### PR DESCRIPTION
**Pull request by msquance-stem**
## Status

* Current Status: Ready for tech review
* Closes https://github.com/NCCE/teachcomputing.org-issues/issues/2587

## Review progress:

- [ ] Browser tested
- [ ] Front-end review completed
- [x] Tech review completed

## What's changed?

Fixing bug due to nil variable

### CODE REVIEW

File: app/jobs/auto_enrol_job.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def perform(achievement:)
 
       Programmes::UserEnroller.new(
         programme_id: programme.id,
-        pathway_slug: pathway.slug,
+        pathway_slug: pathway&.slug,
```
Comment (markjs) [inline app/jobs/auto_enrol_job.rb:23]: This change looks pragmatic and the user enroller is already set up to handle a nil pathway_slug 👍

File: spec/jobs/auto_enrol_job_spec.rb [hunk 2]
```diff
@@ -7,17 +7,19 @@
 
       secondary_certificate = create(:secondary_certificate)
       primary_certificate = create(:primary_certificate)
+      subject_knowledge = create(:cs_accelerator)
```
Comment (markjs) [inline spec/jobs/auto_enrol_job_spec.rb:10]: What is the reasoning for this variable being called `subject_knowledge` while the programme is also referred to as `cs_accelerator`? Are these names analogous? It seems inconsistent with the two lines above to use a different name but this isn't the only place that name is used like that so I don't have a strong feeling either way.

**Review by markjs — APPROVED** [review]
One question but not a deal-breaker 👍

File: spec/jobs/auto_enrol_job_spec.rb (hunk 2 above)
Comment (msquance-stem) [reply spec/jobs/auto_enrol_job_spec.rb:10]: cs_accelerator has become subject_knowledge, and its been changed in some places in but not others, I started out calling it subject knowledge, then realised afterwards it was still called cs_accelerator in the factory and just forgot to update the variable

**Pull request merged**